hack/verify-images-updated.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# Purpose: This script ensures that specific files containing hardcoded release versions are up to date with the current release and do not reference any previous versions.
+
+# Functionality:
+# 1. Determines if the current branch is a release branch (e.g., release-4.x).
+# 2. Extracts the previous version (e.g. 4.16 for release-4.17).
+# 3. Checks specific files for any occurrences of the previous version.
+# 4. If matches are found, it prompts the user to update references to the current release branch and exits with an error code.
+# 5. Skips checks for non-release branches or missing files.
+
```

Comment:
Maybe it worth add
`set -x` to make it easier for debugging in case we'll find an issue with this script